packages/edit-site/src/components/dataviews/in-filter.js

```diff
@@ -2,62 +2,100 @@
  * WordPress dependencies
  */
 import {
-	__experimentalInputControlPrefixWrapper as InputControlPrefixWrapper,
-	SelectControl,
+	Button,
+	privateApis as componentsPrivateApis,
+	Icon,
 } from '@wordpress/components';
+import { check, chevronDown } from '@wordpress/icons';
 import { __, sprintf } from '@wordpress/i18n';
 
+/**
+ * Internal dependencies
+ */
+import { unlock } from '../../lock-unlock';
+
 export const OPERATOR_IN = 'in';
 
+const { DropdownMenuV2, DropdownMenuItemV2, DropdownMenuSeparatorV2 } = unlock(
+	componentsPrivateApis
+);
+
 export default ( { filter, view, onChangeView } ) => {
-	const valueFound = view.filters.find(
-		( f ) => f.field === filter.field && f.operator === OPERATOR_IN
+	const filterInView = view.filters.find( ( f ) => f.field === filter.field );
+	const activeElement = filter.elements.find(
+		( element ) => element.value === filterInView?.value
 	);
 
-	const activeValue =
-		! valueFound || ! valueFound.hasOwnProperty( 'value' )
-			? ''
-			: valueFound.value;
-
-	const id = `dataviews__filters-in-${ filter.field }`;
-
 	return (
-		<SelectControl
-			id={ id }
-			__nextHasNoMarginBottom
-			value={ activeValue }
-			prefix={
-				<InputControlPrefixWrapper
-					as="label"
-					htmlFor={ id }
-					className="dataviews__select-control-prefix"
-				>
-					{ sprintf(
-						/* translators: filter name. */
-						__( '%s:' ),
-						filter.name
-					) }
-				</InputControlPrefixWrapper>
+		<DropdownMenuV2
+			key={ filter.field }
+			label={ filter.name }
+			trigger={
+				<Button variant="tertiary" size="compact">
+					{ activeElement !== undefined
+						? sprintf(
+								/* translators: 1: Filter name. 2: filter value. e.g.: "Author is Admin". */
+								__( '%1$s is %2$s' ),
+								filter.name,
+								activeElement.label
+						  )
+						: filter.name }
+					<Icon icon={ chevronDown } />
+				</Button>
 			}
-			options={ filter.elements }
-			onChange={ ( value ) => {
-				const filters = view.filters.filter(
-					( f ) =>
-						f.field !== filter.field || f.operator !== OPERATOR_IN
+		>
+			{ filter.elements.map( ( element ) => {
+				return (
+					<DropdownMenuItemV2
+						key={ element.value }
+						suffix={
+							activeElement?.value === element.value && (
+								<Icon icon={ check } />
+							)
+						}
```

Comment:
There are dedicated dropdown menu items for checkboxes and radio, which already take care of semantics and check icons — see these two examples for the new `DropdownMenuV2Ariakit` component:
- [checkboxes](https://wordpress.github.io/gutenberg/?path=/story/components-experimental-dropdownmenu-v2-ariakit--with-checkboxes)
- [radios](https://wordpress.github.io/gutenberg/?path=/story/components-experimental-dropdownmenu-v2-ariakit--with-radios)

And related code: https://github.com/WordPress/gutenberg/blob/36617fbee8d6f545deac68c7511735f27792655c/packages/components/src/dropdown-menu-v2-ariakit/stories/index.story.tsx#L149-L307

Reply:
I switched to `DropdownMenuCheckboxItem` at https://github.com/WordPress/gutenberg/pull/56110/commits/0bc02b0155664ae8b7674fd37ce5c24a25603831, thanks.

Tried the new Ariakit ones (https://github.com/WordPress/gutenberg/pull/56110/commits/203fb3fdb9c46d13c761934b0ff436d68ae57b32 and then reverted https://github.com/WordPress/gutenberg/pull/56110/commits/6649de354e6222bcf6fa5f1139f483b0ba2cbc73):

- the radio doesn't allow having a "non-checked" state
- the checkbox performs the same as the current one, though there's a subtle style different when the element is focused, so I think it's best to update all existing dropdowns at once (I can help with this):

| V2 | V2Ariakit |
| --- | --- |
| <img width="234" alt="Captura de ecrã 2023-11-20, às 14 13 24" src="https://github.com/WordPress/gutenberg/assets/583546/a6d7cf5f-0caa-4a11-bc02-6be529ee61f0"> | <img width="243" alt="Captura de ecrã 2023-11-20, às 14 12 06" src="https://github.com/WordPress/gutenberg/assets/583546/308a431a-01c8-461a-95cb-f43394a9e49f"> |

Reply:
Thank you for the feedback!

> the radio doesn't allow having a "non-checked" state

Regarding the radios, could be more specific?
- If you mean that the component can't be initially rendered with no selection, that is probably a bug
- If, instead, you mean that once a radio item is selected, it's not possible to "un-select" it, then that's by design (see this comment for more [info](https://github.com/WordPress/gutenberg/pull/55625#discussion_r1376355788)). The correct approach is then to either switch to checkboxes (in case more than one item can be selected at the same time), or to add a "None" (or any other label) additional menu item.


> the checkbox performs the same as the current one, though there's a subtle style different when the element is focused, so I think it's best to update all existing dropdowns at once

Makes sense. FYI I've been working on [applying the latest design feedback ](https://github.com/WordPress/gutenberg/pull/56041) to the `ariakit` version (the radix version is no longer maintained and scheduled for removal as soon as all usages are migrated)

> (I can help with this)

Thank you!

Reply:
> If, instead, you mean that once a radio item is selected, it's not possible to "un-select" it, then that's by design (see this comment for more https://github.com/WordPress/gutenberg/pull/55625#discussion_r1376355788). The correct approach is then to either switch to checkboxes (in case more than one item can be selected at the same time), or to add a "None" (or any other label) additional menu item.

It was this. Thanks for clarifying how it works. I've lost track in which PR/issue I saw this mockup, but I saw something like this:

- filters that allow multiselection would use a check
- filters that only allow a single item being selected would use a radio

Radios working as you mentioned, we need to clarify the expectations for them. I'm now working on adding multi-selection support at https://github.com/WordPress/gutenberg/pull/56468 so I'll ask design there (will cc you for awareness, but no need to respond!).

Reply:
follow-up https://github.com/WordPress/gutenberg/pull/56468#issuecomment-1824042016